blog/2023-12-19-how-to-keep-mqtt-data-node.mdx

```diff
@@ -0,0 +1,194 @@
+---
+title: "How to Keep a History of MQTT Data With Node.js"
+description: "A tutorial to keep a history of MQTT messages with Node.js by using the ReductStore Client SDK for JavaScript."
+authors: alexey
+tags: [tutorials, iot, mqtt]
+slug: tutorials/iot/how-to-keep-mqtt-data-node
+date: 2023-12-19
+---
+
+The [**MQTT protocol**](https://mqtt.org/) is widely used in IoT applications because of its simplicity and ability to connect different data sources to applications using a publish/subscribe model. While many MQTT brokers support persistent sessions and can store message history as long as an MQTT client is not available, there may be cases where data needs to be stored for a longer period. In such cases, it is recommended to use a time series database. There are many options available, but if you need to store unstructured data such as images, sensor data, or [**Protobuf**](https://protobuf.dev/) messages, consider using [**ReductStore**](https://www.reduct.store). It is a time series database specifically designed for storing large amounts of blob data and optimized for IoT and edge computing.
+
+ReductStore provides client SDKs for many programming languages to integrate it into your infrastructure. In this example, we will use the client SDK for [**JavaScript**](https://github.com/reductstore/reduct-js).
+
+Let’s make a simple MQTT application to see how it works.
+
+{/* truncate */}
+
+## Prerequisites
+
+For this usage example, we have the following requirements:
+
+- Linux AMD64
+- Docker and Docker Compose
+- NodeJS >= 16
+
+If you are an Ubuntu user, you can install the dependencies by running the following command in your terminal:
+
+```
+$ sudo apt-get update
+$ sudo apt-get install docker-compose nodejs
+```
+
+## Run MQTT Broker and ReductStore with Docker Compose
+
+The easiest way to run the broker and the database is to use Docker Compose. So we should create a `docker-compose.yml` file in the example’s folder with the services:
```

Comment:
In the sentence "So we should create a docker-compose.yml file in the example’s folder with the services"

You wanted to link the example folder already here ?

Or is it something like: "The easiest way to run the broker and the database is to use Docker Compose with the docker-compose.yml file that follows."

Reply:
The second variant. I'll change.